### ISSUE TITLE
feat(tensor_parallel): complete ColumnParallelLinear implementation

### ISSUE THREAD

**Issue by dylanyunlon** [dylanyunlon/Neuron_SP#116]
## 问题描述
`deepspeed/core/tensor_parallel/layers.py` 中 `ColumnParallelLinear` 已从 Megatron-LM 迁移框架，需要完成以下 DES-LOC 扩展：
- 异构 head-count padding（非整除 TP，如 32 heads / 5 GPUs）
- `sharded_state_dict()` 接口与 `dist_checkpointing` 框架对接
- `is_expert=True` 路径下 expert-parallel RNG 初始化

## 技术方案
1. 在 `ColumnParallelLinear.__init__` 中计算 `padded_output_size = ceil(output_size / tp_world_size) * tp_world_size`，末 rank 分配 dummy 行并初始化为零
2. 在 `forward()` 中对 dummy 列做 mask（`[:, :output_size]`）保证数值等价
3. 实现 `sharded_state_dict()`：对 weight 在 axis=0 打 TP shard tag，调用 `dist_checkpointing.mapping.ShardedTensor`
4. 连接 `autosp_hook.py` 中的 `register_tp_hooks()`，确保 `desloc_engine` 可通过 `autosp` 路径调用
5. 单元测试文件：`tests/unit/core/tensor_parallel/test_column_parallel_linear.py`

## 验收条件
```bash
# 1. 导入不报错
python -c "from deepspeed.core.tensor_parallel.layers import ColumnParallelLinear; print('OK')"

# 2. 无 NotImplementedError（grep 检查）
grep -n "NotImplementedError" deepspeed/core/tensor_parallel/layers.py | grep -i "column" && echo "FAIL" || echo "PASS"

# 3. heterogeneous padding 单测（32 heads, tp=5）
python -m pytest tests/unit/core/tensor_parallel/test_column_parallel_linear.py -k "hetero" -v

# 4. sharded_state_dict 接口存在
python -c "from deepspeed.core.tensor_parallel.layers import ColumnParallelLinear; import inspect; assert 'sharded_state_dict' in dir(ColumnParallelLinear); print('sharded_state_dict OK')"

# 5. desloc_engine autosp 集成冒烟测试
python -m pytest tests/unit/core/hetero_bridge/test_autosp_hook.py -k "column" -v
```

## 优先级
P1 — tensor parallel 是 DES-LOC 异构训练的基础组件，blocking #126 (hetero_bridge wiring)
